src/array/cpu/segment_reduce.h

```diff
@@ -101,6 +103,58 @@ void ScatterAdd(NDArray feat, NDArray idx, NDArray out) {
   }
 }
 
+/*!
+ * \brief CPU kernel of Scatter Add (on first dimension) operator for heterogeneous graph.
+ * \note math equation: out[idx[i], *] += feat[i, *]
+ * \param feat The input tensor.
+ * \param idx The indices tensor.
+ * \param out The output tensor.
+ */
+
+template <typename IdType, typename DType>
+void UpdateGradMinMax_hetero(HeteroGraphPtr graph,
+                       std::vector<NDArray> list_feat,
+                       std::vector<NDArray> list_idx,
+                       std::vector<NDArray> list_idx_ntypes,
+                       std::vector<NDArray> list_out) {
```

Comment:
So this only handles `copy_u` + `max/min`, right?  Just to make sure.

Reply:
Both `copy_u` and `copy_e`